src/domain/task/helper/test.helper.ts

```diff
@@ -1,22 +1,55 @@
+/* eslint-disable @typescript-eslint/prefer-readonly-parameter-types */
+import type { OrderedSet } from 'immutable'
 import type { Error } from 'domain/error/entity/error'
 import type { EventMetadata } from 'eventBus/eventBus'
-import type { Pair } from 'superTypes'
-import type { Task, UpdateTask } from '../entity/task'
+import type { DeepReadonly, Pair } from 'superTypes'
+import type { Task } from '../entity/task'
+import type { AppState } from '../store/appState'
 
-export type EventParameter = Pair<
-  { type: string; payload: Error | Task | UpdateTask },
-  EventMetadata
->
-type Payload = Error | Task | UpdateTask
+export type EventParameter<T> = Pair<{ type: string; payload: Payload<T> }, EventMetadata>
+type Payload<T> = Error | T
 
-export const getExpectedEventParameter = (
+export const getExpectedEventParameter = <T>(
   type: string,
-  payload: Payload,
+  payload: Payload<T>,
   initiator: string,
   date: Readonly<Date>
-): EventParameter => {
+): EventParameter<T> => {
   return [
     { ...{ type }, ...{ payload } },
     { initiator, timestamp: date }
   ]
 }
+
+// Compute state after invoking removeTask* commands
+export const getExpectedStateAfterRemove = (
+  iniialState: DeepReadonly<AppState>,
+  tasks: DeepReadonly<Task[]>,
+  errorIndex?: number
+): AppState =>
+  tasks.reduce<AppState>((acc: DeepReadonly<AppState>, cur: Task, index: number): AppState => {
+    const foundTask = acc.task.byId.get(cur.id)
+    return index !== errorIndex
+      ? {
+          task: {
+            byId: foundTask ? acc.task.byId.remove(cur.id) : acc.task.byId,
+            byType: acc.task.byType
+              .map((value: Readonly<OrderedSet<string>>) => value.delete(cur.id))
+              .filter((value: Readonly<OrderedSet<string>>) => !value.isEmpty())
+          },
+          displayedTaskIds: acc.displayedTaskIds.remove(cur.id)
+        }
+      : acc
+  }, iniialState)
```

Comment:
```suggestion
tasks.reduce<AppState>((acc: DeepReadonly<AppState>, cur: Task, index: number): AppState => 
  index !== errorIndex
    ? {
        task: {
          byId: acc.task.byId.get(cur.id) ? acc.task.byId.remove(cur.id) : acc.task.byId,
          byType: acc.task.byType
            .map((value: Readonly<OrderedSet<string>>) => value.delete(cur.id))
            .filter((value: Readonly<OrderedSet<string>>) => !value.isEmpty())
        },
        displayedTaskIds: acc.displayedTaskIds.remove(cur.id)
      }
    : acc
, iniialState)
```

It's a suggestion It's up to you

Reply:
@JoeyDjr 
Thanks for you suggestion to be more functional, if you don't mind I will keep the original version as the code is only executed in `test` files, so I think the modification is not relevant here.

Reply:
ok for me